verifier/verifier_test.go

```diff
@@ -605,3 +605,84 @@ func TestVerifyX509TrustedIdentities(t *testing.T) {
+	}
+	opts := notation.VerifyOptions{ArtifactReference: mock.SampleArtifactUri, SignatureMediaType: "application/jose+json"}
+	envelopeMediaType := opts.SignatureMediaType
+	trustPolicy, _ := v.trustPolicyDoc.GetApplicableTrustPolicy(mock.SampleArtifactUri)
+	pluginConfig := opts.PluginConfig
+	verificationLevel, _ := trustPolicy.SignatureVerification.GetVerificationLevel()
+
+	testSignatureArrHigh := [][]byte{semverSigEnv1, semverSigEnv2, semverSigEnv3}
+	testSignatureArrLow := [][]byte{semverSigEnv4, semverSigEnv5, semverSigEnv6}
+
+	for _, testSemSigEnv := range testSignatureArrHigh {
+		outcome := &notation.VerificationOutcome{
+			RawSignature:      testSemSigEnv,
+			VerificationLevel: verificationLevel,
+		}
+		err := v.processSignature(context.Background(), testSemSigEnv, envelopeMediaType, trustPolicy, pluginConfig, outcome)
+		verificationPluginMinVersion, _ := getVerificationPluginMinVersion(&outcome.EnvelopeContent.SignerInfo)
+		outcome.Error = err
+		var errMsg string = "found plugin io.cncf.notary.plugin.unittest.mock with version 1.0.0 but signature verification needs plugin version greater than or equal to " + verificationPluginMinVersion
+		if outcome.Error == nil || outcome.Error.Error() != errMsg {
+			t.Errorf("TestPluginVersionCompatibility Error is %s", outcome.Error.Error())
+		}
+	}
+
+	for _, testSemSigEnv := range testSignatureArrLow {
+		outcome := &notation.VerificationOutcome{
+			RawSignature:      testSemSigEnv,
+			VerificationLevel: verificationLevel,
+		}
+		err := v.processSignature(context.Background(), testSemSigEnv, envelopeMediaType, trustPolicy, pluginConfig, outcome)
+		verificationPluginMinVersion, _ := getVerificationPluginMinVersion(&outcome.EnvelopeContent.SignerInfo)
+		outcome.Error = err
+		var errMsg string = "found plugin io.cncf.notary.plugin.unittest.mock with version 1.0.0 but signature verification needs plugin version greater than or equal to " + verificationPluginMinVersion
+		if outcome.Error.Error() == errMsg {
+			t.Errorf("TestPluginVersionCompatibility Error is %s", outcome.Error.Error())
+		}
+	}
+}
+
+func TestIsRequiredVerificationPluginVer(t *testing.T) {
+
+	var testPlugVer string = "1.0.0"
+
+	testArrLow := []string{"0.0.9", "1.0.0", "1.0.0-alpha", "1-pre+meta"}
+	testArrHigh := []string{"1.0.1", "1.1.0", "1.1.1", "1.2.0"}
+
+	for _, testMinVer := range testArrLow {
+		val := isRequiredVerificationPluginVer(testPlugVer, testMinVer)
+		if !val {
+			t.Errorf("TestVersionCompare Error: Error with version compare function, version %s has to be higher than %s, but the output is lower, function error: %v WantErr: true", testPlugVer, testMinVer, val)
+		}
+	}
+
+	for _, testMinVer := range testArrHigh {
+		val := isRequiredVerificationPluginVer(testPlugVer, testMinVer)
+		if val {
+			t.Errorf("TestVersionCompare Error: Error with version compare function, version %s has to be higher than %s, but the output is lower, function error: %v WantErr: true", testMinVer, testPlugVer, val)
+		}
```

Comment:
Can we have a array with [{0.0.9", "error templete"}] and then iterate over it

Reply:
To keep the tests uniform across the functions in _test.go file, I have used struct instead of map which does not take bool values in the TestIsRequiredVerificationPluginVer function